### PR TITLE
Modify Equals methods on Validator, ValidatorSet

### PR DESCRIPTION
**Pull request by OnedgeLee**
SSIA

### CODE REVIEW

File: Libplanet/Consensus/ValidatorSet.cs [hunk 1]
```diff
@@ -110,18 +110,84 @@ public BigInteger GetValidatorsPower(List<PublicKey> publicKeys)
...
+            => new ValidatorSet(Validators.Add(validator).ToList());
+
+        /// <summary>
+        /// Create new <see cref="ValidatorSet"/> that given validator has been removed.
+        /// Original <see cref="ValidatorSet"/> does not change.
+        /// On removal, check given validator's publickey and power.
+        /// </summary>
+        /// <param name="validator">The validator to be removed.</param>
+        /// <returns>New <see cref="ValidatorSet"/> that given validator has been removed.</returns>
+        public ValidatorSet Remove(Validator validator)
```
Comment (limebell) [inline Libplanet/Consensus/ValidatorSet.cs:169]: Method does not changes instance, can you add `[Pure]` attribute to the method?

File: Libplanet/Consensus/ValidatorSet.cs [hunk 2]
```diff
@@ -110,18 +110,84 @@ public BigInteger GetValidatorsPower(List<PublicKey> publicKeys)
...
+
+        /// <summary>
+        /// Create new <see cref="ValidatorSet"/>
+        /// that validator of given public key has been removed.
+        /// Original <see cref="ValidatorSet"/> does not change.
+        /// </summary>
+        /// <param name="publicKey">The <see cref="PublicKey"/> of validator to be removed.</param>
+        /// <returns>New <see cref="ValidatorSet"/> that
+        /// validator of given public key has been removed.</returns>
+        public ValidatorSet Remove(PublicKey publicKey)
```
Comment (limebell) [inline Libplanet/Consensus/ValidatorSet.cs:180]: Method does not changes instance, can you add `[Pure]` attribute to the method?

File: Libplanet/Consensus/ValidatorSet.cs [hunk 3]
```diff
@@ -110,18 +110,84 @@ public BigInteger GetValidatorsPower(List<PublicKey> publicKeys)
...
+        public bool PublicKeyEquals(ValidatorSet? other) =>
+            other is ValidatorSet validators && PublicKeys.SequenceEqual(validators.PublicKeys);
+
+        /// <summary>
+        /// Create new <see cref="ValidatorSet"/> that given validator has been added.
+        /// Original <see cref="ValidatorSet"/> does not change.
+        /// </summary>
+        /// <param name="validator">The validator to be added.</param>
+        /// <returns>New <see cref="ValidatorSet"/> that given validator has been added.</returns>
+        public ValidatorSet Add(Validator validator)
```
Comment (limebell) [inline Libplanet/Consensus/ValidatorSet.cs:159]: Method does not changes instance, can you add `[Pure]` attribute to the method?

Comment (OnedgeLee) [reply Libplanet/Consensus/ValidatorSet.cs:161]: Resolved with commit amend

File: Libplanet/Consensus/ValidatorSet.cs (hunk 2 above)
Comment (OnedgeLee) [reply Libplanet/Consensus/ValidatorSet.cs:184]: Resolved with commit amend

File: Libplanet/Consensus/ValidatorSet.cs (hunk 1 above)
Comment (OnedgeLee) [reply Libplanet/Consensus/ValidatorSet.cs:172]: Resolved with commit amend

**Review by limebell — APPROVED** [review]

**Review by greymistcube — APPROVED** [review]

**Pull request merged**